redis/src/caching/config.rs

```diff
@@ -0,0 +1,49 @@
+use std::num::NonZeroUsize;
+use std::time::Duration;
+
+/// Defines CacheMode of CacheConfig
+#[derive(Clone, PartialEq, Debug, Copy)]
+pub enum CacheMode {
+    /// Don't cache
+    None,
+    /// Try to cache every command.
+    All,
+    /// Only try to cache commands which explicitly called cache methods.
+    OptIn,
+}
+
+/// Holds configuration to be used by [`crate::caching::CacheManager`]
+#[derive(Clone, Debug, Copy)]
+pub struct CacheConfig {
+    /// `mode` changes the mode of `CacheManager`
+    pub mode: CacheMode,
+    /// `cache_size` defines size of [`crate::caching::cache_manager::LRUCacheShard`].
+    pub cache_size: NonZeroUsize,
+    /// `default_client_ttl` is the maximum amount for holding redis item in the cache.
+    pub default_client_ttl: Duration,
+}
+
+impl CacheConfig {
+    /// Enables cache with 10_000 LRU size and 30 minutes of client ttl
+    pub fn enabled() -> Self {
```

Comment:
this looks like a `Default` implementation. the name doesn't show that this is just a proposed default.

Reply:
With changes to cacheconfig default is now enabled.